experiment_buddy/experiment_buddy.py

```diff
@@ -41,34 +42,20 @@ def register(config_params):
     if hyperparams is not None:
         raise RuntimeError("refusing to overwrite registered parameters")
 
-    for k in config_params.keys():
+    parser = argparse.ArgumentParser()
+    for k, v in config_params.items():
         if k.startswith(wandb_escape):
             raise NameError(f"{wandb_escape} is a reserved prefix")
+        if _valid_hyperparam(k, v):
```

Comment:
Renaming it to _is_hyperparam_valid

Reply:
also since you are now changing this function, add a small documentation between """ doc"""